buildroot/share/PlatformIO/scripts/common-dependencies.py

```diff
@@ -184,7 +184,7 @@ def apply_features_config():
 						src_filter = re.sub(r'[+-]' + d, '', src_filter)
 
 				src_filter = feat['src_filter'] + ' ' + src_filter
-				set_env_field('src_filter', [src_filter])
+				set_env_field('build_src_filter', [src_filter])
```

Comment:
Sorry, I don't know all internals of Marlin. What does mean `set_env_field`?

Shortly about the difference between PlatformIO Core 5 & 6:

1. `src_filter` was renamed to `build_src_filter`. But! You still can refer to any of the old names: `env.GetProjectOption('src_filter') == env.GetProjectOption('build_src_filter')` (backward compatibility)
2. The internal build environment HAS NOT BEEN changed. The new configuration option `build_src_filter` will still be mapped to `SRC_FILTER`. As I said before, a ton of projects use extra scripting and modify `SRC_FILTER` dynamically.

The goal of this "renaming" was to keep config cleaner for the end developer and collect options to their scopes. You can see updated docs where options are grouped => https://docs.platformio.org/en/latest/projectconf/section_env.html#options

Reply:
How about modifying the configuration state in the building environment instead of updating the original config?

Reply:
There might be some technical reason it's done this way —such as collecting items for use in a later step— but if it can be done globally with something like `env.Replace()` (as we now do for `SRC_FILTER`) then we can try that.